### PR TITLE
GIC smp and ls1046a SMP

### PR DESCRIPTION
**Pull request by JiafeiPan**
1. Enable SMP for GIC driver.
2. Add NXP ls1046ardb SMP support.

### CODE REVIEW

**Comment by JiafeiPan** [comment]
@stephanosio and @nashif, would you please review this pull request? thanks.

File: drivers/interrupt_controller/intc_gic.c [hunk 1]
```diff
@@ -93,6 +93,22 @@ void arm_gic_eoi(unsigned int irq)
...
+{
+	uint32_t sgi_val;
+
+	ARG_UNUSED(target_aff);
+
+	sgi_val = GICD_SGIR_TGTFILT_CPULIST |
+		GICD_SGIR_CPULIST(target_list & GICD_SGIR_CPULIST_MASK) |
+		sgi_id;
+
+	__DSB();
```
Comment (carlocaione) [inline drivers/interrupt_controller/intc_gic.c:107]: can you use `dsb()` and `isb()` as defined in `include/arch/arm64/lib_helpers.h` instead?

Comment (JiafeiPan) [reply drivers/interrupt_controller/intc_gic.c:107]: Hi, @carlocaione ,

The GIC driver is both used by ARM and ARM64, I replace __DSB with dsb() and __ISB() with isb(), there is no issue for ARM64, but there will be build warning on ARM platform, for example for board: mercury_xu:

drivers/interrupt_controller/intc_gic.c:107:2: warning: implicit declaration of function 'dsb' [-Wimplicit-function-declaration]
107 | dsb();
| ^~~
drivers/interrupt_controller/intc_gic.c:109:2: warning: implicit declaration of function 'isb' [-Wimplicit-function-declaration]
109 | isb();
| ^~~

but there is no such issue when we use __DSB() and __ISB(),  thanks.

Comment (carlocaione) [reply drivers/interrupt_controller/intc_gic.c:107]: Hoooo, right, I forgot this driver is shared with the ARM boards. Nevermind then.

**Review by carlocaione — APPROVED** [review]

**Review by npitre — APPROVED** [review]

**Pull request merged**